client/assets/javascripts/modules/googleAnalyticsEvents.js

```diff
@@ -0,0 +1,29 @@
+class GoogleAnalyticsEvents {
+  static init () {
+    const elements = document.querySelectorAll('button[data-attribute="ga-event"]')
+
+    elements.forEach(element => {
+      element.addEventListener('click', (e) => { this.sendEvent(e) })
+    })
+  }
+
+  static extractEventInfo (eventElement) {
+    return {
+      action: eventElement.dataset.action,
+      event_params:
+        {
+          event_category: eventElement.dataset.category,
+          event_label: eventElement.dataset.label
+        }
+    }
+  }
+
+  static sendEvent (event) {
+    const eventElement = event.target
+    const eventInfo = this.extractEventInfo(eventElement)
+
+    global.gtag('event', eventInfo.action, eventInfo.event_params)
```

Comment:
I would be tempted to add some defensive code here to check that gtag exists before trying to use it, to avoid async loading issues.

Reply:
Agree - added in. I ended up wrapping the gtag global function in a globals namespace so I could test it. Not sure if there are any problems with this approach? I don't touch JS often enough to know if having globals is even frowned on but it made testing near impossible.